src/test/java/org/opensearch/security/test/helper/rest/RestHelper.java

```diff
@@ -258,27 +272,70 @@ protected final CloseableHttpClient getHTTPClient() throws Exception {
 		return hcb.build();
 	}
 
+	protected final SSLContext setHTTPClientSSL() throws Exception {
+
+		log.debug("Configure HTTP client with SSL");
+
+		if(prefix != null && !keystore.contains("/")) {
+			keystore = prefix+"/"+keystore;
+		}
+
+		final String keyStorePath = FileHelper.getAbsoluteFilePathFromClassPath(keystore).toFile().getParent();
+
+		final KeyStore myTrustStore = KeyStore.getInstance("JKS");
+		myTrustStore.load(new FileInputStream(keyStorePath+"/truststore.jks"),
+				"changeit".toCharArray());
+
+		final KeyStore keyStore = KeyStore.getInstance("JKS");
+		keyStore.load(new FileInputStream(FileHelper.getAbsoluteFilePathFromClassPath(keystore).toFile()), "changeit".toCharArray());
+
+		final SSLContextBuilder sslContextbBuilder = SSLContexts.custom();
+
+		if (trustHTTPServerCertificate) {
+			sslContextbBuilder.loadTrustMaterial(myTrustStore, null);
+		}
+
+		if (sendAdminCertificate) {
+			sslContextbBuilder.loadKeyMaterial(keyStore, "changeit".toCharArray());
+		}
+
+		return sslContextbBuilder.build();
+	}
+
+	protected final CredentialsProvider getClientCredentials() {
+		CredentialsProvider provider = new BasicCredentialsProvider();
+		UsernamePasswordCredentials credentials = new UsernamePasswordCredentials("sarek", "sarek");
+		provider.setCredentials(AuthScope.ANY, credentials);
+		return provider;
+	}
+
 	
 	public class HttpResponse {
-		private final CloseableHttpResponse inner;
+		private final org.apache.http.HttpResponse inner;
 		private final String body;
 		private final Header[] header;
 		private final int statusCode;
 		private final String statusReason;
 
 		public HttpResponse(CloseableHttpResponse inner) throws IllegalStateException, IOException {
+			this((org.apache.http.HttpResponse)inner);
```

Comment:
Why the ctor is necessary?

Reply:
> CloseableHttpAsyncClient.execute() returns Future<org.apache.http.HttpResponse> instead of Future<CloseableHttpResponse>.

mentioned above

Reply:
I still don't see why the ctor is necessary and how the return type of `CloseableHttpAsyncClient.execute()` impacts the ctor that does not do anything other than calling the other ctor anyway.

Reply:
My mistake. This ctor is indeed unnecessary. Fixed.